src/shared/components/textIconArea/TextIconArea.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import _ from 'lodash';
+import ReactDOM from 'react-dom';
+import { observable } from 'mobx';
+import autobind from "autobind-decorator";
+
+export interface ITextIconAreaProps {
+    elements: ITextIconAreaItemProps[];
+    text:string;
+    placeholder?: string;
+    classNames?: string[];
+    onItemRemove?: (d:string) => void;
+    onChange?: (s:string) => string;
```

Comment:
can you make the parameters to onChange and onItemRemove more descriptive? is `d` the value of ITextIconAreaItemProps? (it should be) is `s` the text in the textarea?>

Reply:
I have made the callback names and arguments more clear:
>    onIconClicked?: (itemId:string) => void;
>    onChangeTextArea?: (textAreaContents:string) => string;

Reply:
could you make it `itemValue` so it corresponds with the member name on  `ITextIconAreaItemProps` ? That's what it is, right?

Reply:
You are correct in that it points to the `value` field of the `ITextIconAreaItemProps` interface. I have Updated the name to refect this. Good suggestion.